components/playlist/browser/mime_util.cc

```diff
@@ -0,0 +1,132 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/playlist/browser/mime_util.h"
+
+#include <optional>
+#include <utility>
+
+#include "base/containers/flat_map.h"
+#include "base/files/file_path.h"
+#include "base/no_destructor.h"
+
+namespace playlist {
+namespace {
+
+// References
+// * List of mimetypes registered to IANA
+//   * Video:
+//   https://www.iana.org/assignments/media-types/media-types.xhtml#video
+//   * Audio:
+//   https://www.iana.org/assignments/media-types/media-types.xhtml#audio
+// * Chromium media framework supports
+//   * media/base/mime_util_internal.cc
+// * Mimetype to extension
+//   * https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types
+//
+// Note that methods in "net/base/mime_util.h" have different result.
```

Comment:
Out of interest, what is it that stops us using net/base/mime_util?

Reply:
Yeah, good question. I've compared the outputs and found that net::base::mime_util doesn't return correct info for some media types like mp2t and 3gpp. These seem to be supported by the media framework, so made our own version to cover that.